modules/percolator/src/main/java/org/elasticsearch/percolator/PercolateQueryBuilder.java

```diff
@@ -316,113 +318,52 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
-                        builder.copyCurrentStructure(parser);
-                        builder.flush();
-                        documents.add(BytesReference.bytes(builder));
-                    }
-                } else {
-                    throw new ParsingException(parser.getTokenLocation(), "[" + PercolateQueryBuilder.NAME +
-                            "] query does not support field name [" + currentFieldName + "]");
-                }
-            } else if (token.isValue() || token == XContentParser.Token.VALUE_NULL) {
-                if (QUERY_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
-                    field = parser.text();
-                } else if (NAME_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
-                    name = parser.textOrNull();
-                } else if (INDEXED_DOCUMENT_FIELD_INDEX.match(currentFieldName, parser.getDeprecationHandler())) {
-                    indexedDocumentIndex = parser.text();
-                } else if (INDEXED_DOCUMENT_FIELD_ID.match(currentFieldName, parser.getDeprecationHandler())) {
-                    indexedDocumentId = parser.text();
-                } else if (INDEXED_DOCUMENT_FIELD_ROUTING.match(currentFieldName, parser.getDeprecationHandler())) {
-                    indexedDocumentRouting = parser.text();
-                } else if (INDEXED_DOCUMENT_FIELD_PREFERENCE.match(currentFieldName, parser.getDeprecationHandler())) {
-                    indexedDocumentPreference = parser.text();
-                } else if (INDEXED_DOCUMENT_FIELD_VERSION.match(currentFieldName, parser.getDeprecationHandler())) {
-                    indexedDocumentVersion = parser.longValue();
-                } else if (AbstractQueryBuilder.BOOST_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
-                    boost = parser.floatValue();
-                } else if (AbstractQueryBuilder.NAME_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
-                    queryName = parser.text();
-                } else {
-                    throw new ParsingException(parser.getTokenLocation(), "[" + PercolateQueryBuilder.NAME +
-                            "] query does not support [" + currentFieldName + "]");
-                }
-            } else {
-                throw new ParsingException(parser.getTokenLocation(), "[" + PercolateQueryBuilder.NAME +
-                        "] query does not support [" + token + "]");
-            }
-        }
-
-        PercolateQueryBuilder queryBuilder;
-        if (documents.isEmpty() == false) {
-            queryBuilder = new PercolateQueryBuilder(field, documents, XContentType.JSON);
-        } else if (indexedDocumentId != null) {
-            queryBuilder = new PercolateQueryBuilder(field, indexedDocumentIndex, indexedDocumentId, indexedDocumentRouting,
-                indexedDocumentPreference, indexedDocumentVersion);
+    @SuppressWarnings("unchecked")
```

Comment:
I tend to put this on the variable that needs it for extra paranoia.

Reply:
TIL you can do this on variables - thanks!